typegen/src/lib.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright 2019-2022 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
```

Comment:
Let's add a `#![deny(unused_crate_dependencies)]` here, if not done elsewhere, to double check our dependencies

Reply:
That doesn't always work well but def worth a go!

And `#![deny(missing_docs)]` too please to make sure we document the public interface :)